### PR TITLE
Add examples and explanations to several item docs

### PR DESCRIPTION
**Pull request by torkleyy**
Partially addresses #145

### CODE REVIEW

File: src/world.rs [hunk 1]
```diff
@@ -722,6 +774,37 @@ impl World {
...
+    ///
+    /// There are two built-in resources:
+    ///
+    /// * `LazyUpdate` and
+    /// * `EntitiesRes`
+    ///
+    /// Both of them should only be fetched immutably, which is why
+    /// the latter one has a type def for convenience: `Entities` which
+    /// is just `Fetch<EntitiesRes>`. Both resources are special and need
+    /// to execute code at the end of the frame, which is done in `World::maintain`.
```
Comment (WaDelma) [inline src/world.rs]: The text is fine, but it reminds me the fact that calling `World::maintain` is not enforced. Is there any way to encode this in type system?

Comment (torkleyy) [reply src/world.rs]: I wouldn't know how, unless we change our architecture :(

**Comment by torkleyy** [comment]
@WaDelma Thanks for your review, I really appreciate that! Regarding the above comments, I have several ideas how to improve Specs, however most of them require specialization or would introduce massive boilerplate otherwise.

bors +

**Comment by torkleyy** [comment]
Oops

bors r+

**Pull request merged**